virttest/virt_vm.py

```diff
@@ -1216,8 +1216,11 @@ def print_guest_network_info():
                 if serial:
                     break
                 raise
-            except remote.LoginProcessTerminatedError:
-                raise
+            except remote.LoginProcessTerminatedError as err:
+                if not re.match(r".*ssh:.*No route", err.output):
+                    raise
+                else:
+                    error = err
```

Comment:
Well as I mentioned the `No route` is not the only message `ssh` reports when the service is still not fully started. There could be `exchange ...` errors or other failures so I don't think we should be raising exceptions.

The more I think about it the more I believe the solutions should be:

```diff
diff --git a/virttest/virt_vm.py b/virttest/virt_vm.py
index fc5d4831..88fd27af 100644
--- a/virttest/virt_vm.py
+++ b/virttest/virt_vm.py
@@ -1217,6 +1217,7 @@ class BaseVM(object):
                     break
                 raise
             except Exception as err:
+                time.sleep(1)
                 error = err
             not_tried = False
```

as that should limit the rate of failed messages and in worst case could decrease the `timeout` of 1s.

Reply:
I mean don't forget that first NetworkManager starts, gets IP address and only then sshd is started. Even that can take quite a while before sshd is fully started (especially when it generates ssh keys) so the overlap between VM having ip address and VM being ready to accept ssh connection is definitely not zero.

Yes, we could come up with limit of the same messages or limit of whatever, but the `time.sleep` should be relatively safe and very simple to understand...

Reply:
(What I meant to say is, perfect solution could include:

```python
timeout_alive = 10
timeout_route = 360
other_err_iterations = 10
while time.time() < end_time or not_tried:
    try:
        return self.login(...)
    except Authentication:
        raise
    except Exception as err:
        if looks_like_route:
            if deadline_route:
                if time.time() > deadline_route:
                    raise DeadlineNoRoute(...)
            else:
                deadline_route = time.time() + timeout_route
        elif deadline_alive:
            if not self.is_alive():
                if time.time() > deadline_alive:
                    raise DeadlineAlive(...)
            else:
                deadline_alive = 0
        else:
            other_err += 1
            if other_errr > other_err_iterations:
                raise OtherError(...)
            time.sleep(1)
```

But that is rather complex and might be worse than just adding the sleep...